helper/pluginutil/logger.go

```diff
@@ -0,0 +1,158 @@
+package pluginutil
+
+import (
+	"bytes"
+	"fmt"
+	stdlog "log"
+	"strings"
+
+	hclog "github.com/hashicorp/go-hclog"
+	log "github.com/mgutz/logxi/v1"
+)
+
+// pluginLogFaker is a wrapper on logxi.Logger that
+// implements hclog.Logger
+type hclogFaker struct {
+	logger log.Logger
+
+	name    string
+	implied []interface{}
+}
+
+func (f *hclogFaker) buildLog(msg string, args ...interface{}) (string, []interface{}) {
+	if f.name != "" {
+		msg = fmt.Sprintf("%s: %s", f.name, msg)
+	}
+	args = append(f.implied, args...)
+
+	return msg, args
+}
+
+func (f *hclogFaker) Trace(msg string, args ...interface{}) {
+	msg, args = f.buildLog(msg, args...)
+	f.logger.Trace(msg, args...)
+}
+
+func (f *hclogFaker) Debug(msg string, args ...interface{}) {
+	msg, args = f.buildLog(msg, args...)
+	f.logger.Debug(msg, args...)
+}
+
+func (f *hclogFaker) Info(msg string, args ...interface{}) {
+	msg, args = f.buildLog(msg, args...)
+	f.logger.Info(msg, args...)
+}
+
+func (f *hclogFaker) Warn(msg string, args ...interface{}) {
+	msg, args = f.buildLog(msg, args...)
+	f.logger.Warn(msg, args...)
+}
+
+func (f *hclogFaker) Error(msg string, args ...interface{}) {
+	msg, args = f.buildLog(msg, args...)
+	f.logger.Error(msg, args...)
+}
+
+func (f *hclogFaker) IsTrace() bool {
+	return f.logger.IsTrace()
+}
+
+func (f *hclogFaker) IsDebug() bool {
+	return f.logger.IsDebug()
+}
+
+func (f *hclogFaker) IsInfo() bool {
+	return f.logger.IsInfo()
+}
+
+func (f *hclogFaker) IsWarn() bool {
+	return f.logger.IsWarn()
+}
+
+func (f *hclogFaker) IsError() bool {
+	return !f.logger.IsTrace() && !f.logger.IsDebug() && !f.logger.IsInfo() && !f.IsWarn()
+}
+
+func (f *hclogFaker) With(args ...interface{}) hclog.Logger {
+	var nf = *f
```

Comment:
I went with this since hclog was doing the same, unless it applies differently in that case: https://github.com/hashicorp/go-hclog/blob/master/int.go#L346